bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,913 @@
+    let timeEntries = [];
+    let today = moment().format('YYYY-MM-DDT00:00:00')
+    let sevenDaysBefore = moment().subtract(7,'d').format('YYYY-MM-DDT00:00:00')
+    let thirtyDaysBefore = moment().subtract(30,'d').format('YYYY-MM-DDT00:00:00')
+
+    let dateFilter = `TimeSheetDate ge datetime'${thirtyDaysBefore}' and TimeSheetDate le datetime'${today}'`
+
+    if (latestSynchronizationTime) {
+        dateFilter = `TimeSheetDate ge datetime'${sevenDaysBefore}'`
+    }
+
+    let statusFilter = "(TimeSheetStatus eq '10' or " +
+        "TimeSheetStatus eq '20' or " +
+        "TimeSheetStatus eq '30' or " +
+        "TimeSheetStatus eq '40' or " +
+        "TimeSheetStatus eq '50')"
+
+    let filter = `${dateFilter} and ${statusFilter}`
+
+    let url = `API_MANAGE_WORKFORCE_TIMESHEET/TimeSheetEntryCollection?$filter=${filter}`
+
+    console.log('## URL -> ' + url)
+
+    let respGET = await client.fetch(url, defaultGetOptions);
+
+    if (!respGET.ok) {
+        throw new Error(`Could not Retrieve Time Entries (${respGET.status}: ${respGET.statusText})`);
+    }
+
+
+    let timeEntryResponse = await respGET.json()
+
+    for (let i = 0; i < timeEntryResponse.d.results.length; i++) {
+
+        let timeEntry = timeEntryResponse.d.results[i]
+
+        let TimeSheetDateTimeStamp =
+            timeEntry.TimeSheetDate.replace(/[^.\d]/g, '');
+
+        timeEntries.push({
+            id: parseInt(timeEntry.TimeSheetRecord),
+            ts_id: timeEntry.TimeSheetRecord,
+            company_code: timeEntry.CompanyCode,
+            ts_date: new Date(parseInt(TimeSheetDateTimeStamp)),
+            person_work_agreement: timeEntry.PersonWorkAgreement,
+            person_work_agreement_external_id: timeEntry.PersonWorkAgreementExternalID,
+            ts_df_controlling_area: timeEntry.TimeSheetDataFields.ControllingArea,
+            ts_df_hours_unit_of_measure: timeEntry.TimeSheetDataFields.HoursUnitOfMeasure,
+            ts_df_purchase_order_item: timeEntry.TimeSheetDataFields.PurchaseOrderItem,
+            ts_df_receiver_cost_center: timeEntry.TimeSheetDataFields.ReceiverCostCenter,
+            ts_df_recorded_hours: timeEntry.TimeSheetDataFields.RecordedHours,
+            ts_df_recorded_quantity: timeEntry.TimeSheetDataFields.RecordedQuantity,
+            ts_df_ts_overtime_category: timeEntry.TimeSheetDataFields.TimeSheetOvertimeCategory,
+            ts_df_ts_task_component: timeEntry.TimeSheetDataFields.TimeSheetTaskComponent,
+            ts_df_ts_task_level: timeEntry.TimeSheetDataFields.TimeSheetTaskLevel,
+            ts_df_ts_task_type: timeEntry.TimeSheetDataFields.TimeSheetTaskType,
+            ts_df_ts_note: timeEntry.TimeSheetDataFields.TimeSheetNote,
+            ts_operation: timeEntry.TimeSheetOperation,
+            ts_status: timeEntry.TimeSheetStatus,
+        })
+    }
+
+    dataStore.save("TimeEntries", timeEntries);
+    dataStore.save("TimeEntriesClock", timeEntries);
```

Comment:
Because of this scenario:
 1 - We need a detail page from TimeEntries with a filter (user email + recorded hours = 0.03).
 2 - We need a table component pointing to TimeEntries to display all entries related to the user (user email)

In this case the logic will be applied also to the table component (The way citrix queries the page causes it) and we don`t want that. That`s the reason we need to store in two different tables.